reporter-influxdb/src/test/java/io/ultrabrew/metrics/reporters/influxdb/InfluxDBReporterTest.java

```diff
@@ -64,6 +64,20 @@ public void testSetBufferSize() {
     ByteBuffer buffer = Deencapsulation.getField(c, "byteBuffer");
     assertEquals(12765, buffer.capacity());
   }
+  
+  @Test
+  public void testSeEndpoint() {
+    InfluxDBReporter r = InfluxDBReporter.builder()
+        .withBaseUri(TEST_URI)
+        .withDatabase("test") // ignored
+        .withEndpoint("/my/change?db=foo")
+        .withBufferSize(12765)
+        .build();
+
+    InfluxDBClient c = Deencapsulation.getField(r, "dbClient");
+    URI uri = Deencapsulation.getField(c, "dbUri");
```

Comment:
Please don't add dependency on `Deencapsulation` since it's deprecated and removed from newer versions of `jmockit`.

Reply:
I was just using what's already there in the other tests. Is there a different way to get to the private state?

Reply:
No, and there should not be; verifying private state is not good testing practice, tests should verify externally visible behavior.

I know some of the old tests do funky things, and plan to fix those, so I'd prefer if you don't add more such cases.

Reply:
In this case, what we want to test is that setting the endpoint actually changes the URI being called, so you'd want to mock the HTTP client and verify that it is called to perform a request to the correct URI.

Reply:
done for this test. The others still need protected getters for their private values in the client.